### PR TITLE
Update matrix_block.js to ts 

### LINKED ISSUE #48: Convert Matrix code to TypeScript
Not important anytime soon, just creating an issue so we dont forget

### PR DESCRIPTION
**Pull request by veritem**
fix #48 

## Types of Changes

-   [ ] Bugfix
-   [ ] Enhancement/optimization
-   [ ] New feature (minor)
-   [ ] New feature (major)
-   [ ] Other

## Issues Fixed or Closed by This PR

-

## Checklist

-   [ ] I have made sure the project builds and runs on my local machine.
-   [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).
-   [ ] My change requires a change to the documentation.
-   [ ] I have updated the documentation accordingly.
-   [ ] I have coordinated this change with the Kobra team.
-   [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
-   [ ] I have added tests to cover my changes.

## Licensing

-   [x] I agree to license my changes under the [new BSD license](https://en.wikipedia.org/wiki/BSD_licenses#3-clause_license_\(%22BSD_License_2.0%22,_%22Revised_BSD_License%22,_%22New_BSD_License%22,_or_%22Modified_BSD_License%22\)).

### CODE REVIEW

**Comment by Merlin04** [comment]
I'm getting a compile error when I try to run this:

```
./src/blocks/matrix_block.ts:244:21
Syntax error: Identifier 'i' has already been declared.

  242 |             }
  243 |             const matrixDataLine = matrixData[j] ?? [];
> 244 |             for (var i = 0; i < this.dimX; i++) {
      |                      ^
  245 |                 this.line[j].appendField(
  246 |                     new Blockly.FieldNumber(matrixDataLine[i] ?? 0),
  247 |                     "element_" + j + i
```

Can you go through the code and fix compile errors until it successfully runs? After that I can review the PR

**Comment by Merlin04** [comment]
Also keep in mind that arrow functions and regular functions aren't equivalent: https://stackoverflow.com/a/34361380

You can use this to get rid of the code with the pattern `var that = this;` as you can directly access `this` in the arrow functions.

**Comment by veritem** [comment]
@Merlin04  I'm not sure if this is the best way to do it. I've tried to find ways to do it better from different source but seems to find nothing helpful

PS: I've also checked https://github.com/google/blockly-samples/tree/master/plugins  

I feel like this should be merged!

File: src/components/GQLUtils.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export interface UseQueryData<T> {
```
Comment (pranavnt) [inline src/components/GQLUtils.ts:1]: There don't appear to be any other references to UseQueryData: 
https://sourcegraph.com/github.com/kobra-dev/Kobra@9687f4ee704ecb30087ca4896fa018b0900baeef/-/blob/src/components/GQLUtils.ts?L1:18#tab=references

Comment (veritem) [reply src/components/GQLUtils.ts:1]: Should be removed?

**Comment by Merlin04** [comment]
I think this branch has fallen too out of date and would be very difficult to merge; I'm going to close this PR and recreate it with only the new `matrix_block.js`

**Pull request closed without merging**